### PR TITLE
Adding Go template support

### PR DESCRIPTION
**Pull request by wgrzelak**
Added Go template support, and `--var` parameter that can be used for providing custom values to the test suite in the executing time. Also env variables can be used.

This PR will fix an issue supported in https://github.com/GoogleCloudPlatform/marketplace-testrunner/issues/5.

Current test suite (with `envsubst`):

```yaml
- name: Site address should be 200 OK
  httpTest:
    url: http://${EXTERNAL_IP}
    expect:
      statusCode:
        equals: 200
      bodyText:
        html:
          title:
            exactly: 'Web Server Application'
```

New test suite (without `envsubst`):

```yaml
- name: Site address should be 200 OK
  httpTest:
    url: http:/{{ .Env.EXTERNAL_IP }}
    expect:
      statusCode:
        equals: 200
      bodyText:
        html:
          title:
            exactly: 'Web Server Application'
```

### CODE REVIEW

**Comment by khajduczenia** [comment]
A great work, Wojtek! Thanks for doing this!

File: specs/specs.go [hunk 1]
```diff
@@ -98,14 +101,32 @@ type StringAssert struct {
...
+func getEnvs() map[string]string {
+	envs := make(map[string]string)
+	for _, element := range os.Environ() {
+		splits := strings.Split(element, "=")
+		envs[splits[0]] = splits[1]
+	}
+	return envs
+}
+
+func LoadSuite(path string, vars *map[string]string) *Suite {
```
Comment (huyhg) [inline specs/specs.go]: this should be `vars map[string]string`, dropping the pointer

File: runner/main.go [hunk 2]
```diff
@@ -73,7 +74,7 @@ func main() {
 	status := testStatus{}
 	for _, testSpec := range *testSpecs {
 		glog.Infof(">>> Running %v", testSpec)
-		suite := specs.LoadSuite(testSpec)
+		suite := specs.LoadSuite(testSpec, vars)
```
Comment (huyhg) [inline runner/main.go]: `*vars`, after changing the parameter type to a map

File: specs/specs.go [hunk 3]
```diff
@@ -98,14 +101,32 @@ type StringAssert struct {
...
-	data, err := ioutil.ReadFile(path)
+type TemplateSpec struct {
+	Vars map[string]string
+	Env  map[string]string
+}
+
+func getEnvs() map[string]string {
+	envs := make(map[string]string)
+	for _, element := range os.Environ() {
+		splits := strings.Split(element, "=")
```
Comment (huyhg) [inline specs/specs.go]: `SplitN(element, "=", 2)`, to properly handle case where `=` is the value content.

File: specs/specs.go [hunk 4]
```diff
@@ -98,14 +101,32 @@ type StringAssert struct {
...
+		envs[splits[0]] = splits[1]
+	}
+	return envs
+}
+
+func LoadSuite(path string, vars *map[string]string) *Suite {
+	templateFile, err := ioutil.ReadFile(path)
 	check(err)
 
+	env := getEnvs()
```
Comment (huyhg) [inline specs/specs.go]: for testability and encapsulation, this should be passed as a parameter to this function as well

**Comment by wgrzelak** [comment]
I've renamed `--vars` param to `--var`. I think it make more sense comparing with `{{ .Env.NAME }}`.

**Pull request merged**